lib/utils.js

```diff
@@ -252,10 +252,16 @@ exports.files = function (dir, ext, ret) {
     .filter(ignored)
     .forEach(function (path) {
       path = join(dir, path);
-      if (statSync(path).isDirectory()) {
-        exports.files(path, ext, ret);
-      } else if (path.match(re)) {
-        ret.push(path);
+      try {
```

Comment:
Hi!  Thanks for the PR.

I'm not sure I agree with the strategy here.  If we want to provide "expected" behavior--and we should--then a broken symlink should still be watched. If/when the symlink is fixed while watching, and the symlink is to a directory, we'd want to watch files in *that* directory as well (if requested).  What happens instead is that broken symlinks silently fail, which causes the same problem for users that this PR was meant to address: "it's not working, and I don't know why".

But that implementation sounds like a pain, and I'm not really interested in significant efforts against the `--watch` functionality as part of Mocha core short of low-severity bugfixes.

Instead of trying to further enhance `--watch`, I'd be happy with a solution which leverages `fs.lstatSync()` and re-throws with a helpful error if a symlink is broken.  That'd let the user know something is wrong, and that they'll need to address it if they want Mocha to do what they're telling it to do.